Sources/StreamChat/Workers/MessageUpdater.swift

```diff
@@ -63,14 +63,10 @@ class MessageUpdater: Worker {
 
             // Hard Deleting is necessary for bounced messages, since these messages are never stored on the cloud
             // an apiClient request to delete them would never be triggered.
-            let shouldBeHardDeleted = hard || messageDTO.failedToBeSentDueToModeration
-            let shouldAllowLocallyStoredMessagesToBeDeleted = !isLocalStorageEnabled
```

Comment:
We aren't missing a case from here, right? An explanation would be nice.

Reply:
No, the previous logic was super confusing. Right now it is pretty simple, and the tests should be self-explanatory as well as the code 🤔 We should hard delete a message if `hard` is equal to `true` or if the message is local only.

Reply:
I will update the comment tho 👍